src/global_logic.js

```diff
@@ -33,6 +33,11 @@ module.exports.isCosmos = function (arm) {
         (skilltypes[arm.skill3] != undefined && skilltypes[arm.skill3].type == "cosmosArm");
 };
 
+module.exports.isDarkOpus = function (arm) {
+    return (arm.skill1 != undefined && arm.skill1 == "normalKamuiL") ||
+        (arm.skill1 != undefined && arm.skill1 == "magnaKamuiL");
+};
+
```

Comment:
`arm.skill1 != undefined` double checking
- [ ] `arm.skill1 != undefined && arm.skill1.endsWith("KamuiL")`
  - but "exKamuiL" may hit this, regex for more strict check.
- [ ] `arm.skill1 != undefined && arm.skill1.match(/^(normal|magna)KamuiL$/);`
  - generate match object unnecessary
- [x] `arm.skill1 != undefined && ((arm.skill1 == "normalKamuiL") || (arm.skill1 == "magnaKamuiL"))`
  - simple is good

Motocal Weapons form allow users to change the skill.
- Users can set Kamui(L) to another weapons.
- in future, if another kamui(L) weapon was added.

Current isDarkOpus work for default DarkOpus weapon,
but no support custom weapons. This can be future topic, new weapon may break this.

Reply:
I added id based comparison on my last commit. That should fix all problems.
New method:
module.exports.isDarkOpus = function (arm) {
    return darkOpusWeapons.some(function (value) {
        return arm.id == value;
    });
};
Dark opus weapons:
var darkOpusWeapons = [
    "1040017000",
    "1040017100",
    "1040212500",
    "1040212600",
    "1040310600",
    "1040310700",
    "1040415000",
    "1040415100",
    "1040809400",
    "1040809500",
    "1040911000",
    "1040911100"
];

ID is parsed from image_url with regex.